Change warning behavior in read_hdf5

matteobachetti opened the pull request:
Resolves #7725

bsipocz commented:
Maybe a bit of an overkill, but can we add a test to read in such a file with a single table and no path?

matteobachetti commented:
I added one test and modified another. I added a catch_warnings call to `test_read_nopath` and verified that there is _no_ warning thrown with that specific message. Then I added `test_read_nopath_multi_tables` where I create a dataset with multiple tables and check that the warning is actually thrown.

matteobachetti commented:
Ping @astrofrog

astrofrog commented on CHANGES.rst:
```diff
@@ -44,6 +44,8 @@ astropy.io.misc
 
 - Implement serialization of ``SkyCoord`` in ASDF. [#8284]
 
+- No warnings when reading HDF5 files with only one table and not ``path=`` argument [#8483]
```
not -> no

matteobachetti commented:
@astrofrog done

bsipocz commented:
This has been already approved, so I'm merging.

Thanks @matteobachetti!
